junebug/plugins/nginx/plugin.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import subprocess
+from os import path, remove
+from distutils.dir_util import mkpath
+from urlparse import urljoin
+
+from confmodel import Config
+from confmodel.fields import ConfigText
+
+from junebug.plugin import JunebugPlugin
+from junebug.utils import channel_public_http_properties
+
+
+log = logging.getLogger(__name__)
+DIRNAME = path.dirname(__file__)
+
+
+class NginxPluginConfig(Config):
+    '''Config for :class:`NginxJunebugPlugin`'''
+
+    vhost_file = ConfigText(
+        "The file to write the junebug nginx vhost file to",
+        default='/etc/nginx/sites-enabled/junebug.conf', static=True)
+
+    locations_dir = ConfigText(
+        "The directory to write location block config files to",
+        default='/etc/nginx/includes/junebug/', static=True)
+
+    server_name = ConfigText(
+        "Server name to use for nginx vhost",
+        required=True, static=True)
+
+    vhost_template = ConfigText(
+        "Path to the template file to use for the vhost config",
+        default='%s/vhost.template' % (DIRNAME,), static=True)
+
+    location_template = ConfigText(
+        "Path to the template file to use for each channel's location config",
+        default='%s/location.template' % (DIRNAME,), static=True)
+
+
+class NginxPlugin(JunebugPlugin):
+    '''
+    Manages an nginx virtual host that proxies to the Junebug instance's
+    http-based channels.
+    '''
+
+    def start_plugin(self, config, junebug_config):
+        self.configured_channels = set()
+        self.config = NginxPluginConfig(config)
+        self.vhost_template = read(self.config.vhost_template)
+        self.location_template = read(self.config.location_template)
+        write(self.config.vhost_file, self.get_vhost_config())
+
+    def stop_plugin(self):
+        ensure_removed(self.config.vhost_file)
```

Comment:
Should we not also remove all of the configured channels? If not all of the same channels get started up when the plugin gets started again, then we will have some locations leading to nowhere.